sync/src/relayer/compact_block_process.rs

```diff
@@ -43,22 +43,25 @@ impl<'a, CS: ChainStore + 'static> CompactBlockProcess<'a, CS> {
         let compact_block: CompactBlock = (*self.message).try_into()?;
         let block_hash = compact_block.header.hash().to_owned();
 
-        if self.relayer.state.already_known_compact_block(&block_hash) {
-            debug!(target: "relay", "discarding already known compact block {:x}", block_hash);
+        let parent = self
```

Comment:
prefer `match` to `unwrap` here:
```
match self.relayer.shared.get_header_view(compact_block.header.parent_hash()) {
    None => {...},
    Some(parent) =>  {...}
}
```

Reply:
Already check `parent.is_none()` below. So here `unwrap` is safe.